vllm/v1/core/sched/scheduler.py

```diff
@@ -436,6 +439,23 @@ def schedule(self) -> SchedulerOutput:
                         if num_new_tokens == 0:
                             # The request cannot be scheduled.
                             break
+                        if self.is_encoder_decoder:
+                            # For encoder-decoder models, we allocate slots for
+                            # the cross-attention blocks based on the max
+                            # encoder length. This is a single static allocation
+                            # and does not grow with the number of decoder
+                            # tokens.
+                            max_encoder_len = MULTIMODAL_REGISTRY.\
```

Comment:
shouldn't this value be dynamic based on the actual input size for the encoder branch?

e.g. for Whisper, if we get a 5s clip rather than 30s clip, do we need to allocate for the full 30s?

cc @NickLucche